src/test/java/com/mm/moneymanager/controller/IncomeControllerTest.java

```diff
@@ -170,4 +171,43 @@ void testSaveIncome() throws Exception {
                 .andReturn();
     }
 
+    @Test
+    @WithMockUser(username = "testuser")
+    void testSaveMalformedIncome() throws Exception {
+        //given
+
+        //when
+        mvc.perform(MockMvcRequestBuilders.post("/api/v1/income/userincome")
+                .accept(MediaType.APPLICATION_JSON)
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(jsonContentIncomeMalformed))
+                //then
+                .andExpect(status().isBadRequest())
+                .andExpect(MockMvcResultMatchers.content().string("{\"recurringIncome\":\"must not be null\",\"incomeSource\":\"income source is mandatory\",\"paymentDate\":\"must not be null\"}"))
+                .andDo(print())
+                .andReturn();
+    }
+
+
+    @Test
+    @WithMockUser(username = "testuser")
+    void testUpdateIncome() throws Exception {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/alex_113/Money-Manager/pullRequest?prid=5653039)